pyprt/pyprt_arcgis/pyprt_arcgis.py

```diff
@@ -41,26 +41,64 @@ def arcgis_to_pyprt(feature_set):
     for feature in feature_set.features:
         try:
             geo = Geometry(feature.geometry)
-            if geo.type == 'Polygon':
-                coord = geo.coordinates()[0]
-                coord_remove_last = coord[:-1]
-                coord_inverse = np.flip(coord_remove_last, axis=0)
-
-                if coord.shape[1] == 2:  # we have to add a dimension
-                    coord_inverse[:, 1] *= -1
-                    coord_add_dim = np.insert(coord_inverse, 1, 0, axis=1)
-                    coord_fin = np.reshape(
-                        coord_add_dim, (1, coord_add_dim.shape[0]*coord_add_dim.shape[1]))
-                elif coord.shape[1] == 3:  # need to swap the 1 and 2 columns
-                    coord_inverse[:, 1] *= -1
-                    coord_swap_dim = coord_inverse.copy()
-                    temp = np.copy(coord_swap_dim[:, 1])
-                    coord_swap_dim[:, 1] = coord_swap_dim[:, 2]
-                    coord_swap_dim[:, 2] = temp
-                    coord_fin = np.reshape(
-                        coord_swap_dim, (1, coord_swap_dim.shape[0]*coord_swap_dim.shape[1]))
-
-                initial_geometry = pyprt.InitialShape(coord_fin.tolist()[0])
+            if geo.type == 'Polygon' and (not geo.is_empty):
+                pts_cnt = 0
+                vert_coord_list = []
+                face_count_list = []
+                holes_ind_list = []
+                coord_list = geo.coordinates()
+
+                for face_idx, coord_part in enumerate(coord_list):
+                    in_geo = Geometry({"rings": [coord_part]})
+                    store_area = in_geo.area
+                    coord_remove_last = coord_part[:-1]
+                    coord_inverse = np.flip(coord_remove_last, axis=0)
+
+                    if len(coord_part[0]) == 2:  # we have to add a dimension

```

Comment:
sub-functionality like "add a dimension" should be extracted into a helper function - the whole function is a bit long